pkgs/development/python-modules/pulumi/default.nix

```
@@ -0,0 +1,101 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, protobuf
+, dill
+, grpcio
+, pulumi-bin
+, isPy27
+, semver
+, pyyaml
+, six
+
+# for tests
+, tox
+, go
+, pulumictl
+, bash
+, pylint
+, pytest
+, pytest-timeout
+, coverage
+, black
+, wheel
+, pytest-asyncio
+
+, mypy
+}:
+
+buildPythonPackage rec {
+  pname = "pulumi";
+  version = "3.25.1";
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "teto";
+    repo = "pulumi";
+    rev = "073e94a0b8b4ef0b1b856c63670a8dd88f6b6d02";
+    sha256 = "sha256-oyjQW/Z1NvsHpUwikX+bl1npfF4LESOua/o1qjqAgUs=";
+  };
+
+  propagatedBuildInputs = [
+    semver
+    protobuf
+    dill
+    grpcio
+    pyyaml
+    six
+  ];
+
+  checkInputs = [
+    pulumi-bin
+    pulumictl
+    mypy
+    bash
+    go
+    tox
+    pylint
+    pytest
+    pytest-timeout
+    coverage
+    pytest-asyncio
+    wheel
+    black
+  ];
+
+  # look into scripts/get-version
+  PULUMI_VERSION=version;
+  SHELL = "${bash}/bin/bash";
+
+  postConfigure = ''
+    cp README.md sdk/python/lib
+    patchShebangs .
+    cd sdk/python/lib
+    substituteInPlace ../Makefile \
+      --replace '$(shell cd ../../ && pulumictl get version)' '${pulumi-bin.version}' \
+      --replace '$(shell cd ../../ && pulumictl get version --language python)' '${version}'
+
+    substituteInPlace ../requirements.txt \
+      --replace 'pylint==2.10.2' 'pylint>=2.10.2'
```

```suggestion
      --replace 'pylint==2.10.2' ''
```
Since we don't want to use pylint anyway

---

best to keep when developing the plugin